base_user_role/tests/test_user_role.py

```diff
@@ -105,3 +105,61 @@ def test_role_1_2_with_dates(self):
         role1_group_ids.append(role1.group_id.id)
         role_group_ids = sorted(set(role1_group_ids))
         self.assertEqual(user_group_ids, role_group_ids)
+
+    def test_role_unlink(self):
+        # Get role1 groups
+        role1 = self.role_model.browse(self.cr, self.uid, self.role1_id)
+        role1_group_ids = role1.implied_ids.ids
+        role1_group_ids.append(role1.group_id.id)
+        role1_group_ids = sorted(set(role1_group_ids))
+        # Get role2 groups
+        role2 = self.role_model.browse(self.cr, self.uid, self.role2_id)
+        role2_group_ids = role2.implied_ids.ids
+        role2_group_ids.append(role2.group_id.id)
+        role2_group_ids = sorted(set(role2_group_ids))
+        # Configure the user with role1 and role2
+        self.user_model.write(
+            self.cr, self.uid, [self.user_id],
+            {'role_line_ids': [
+                (0, 0, {'role_id': self.role1_id}),
+                (0, 0, {'role_id': self.role2_id}),
+            ]})
+        user = self.user_model.browse(self.cr, self.uid, self.user_id)
+        # Remove role2
+        role2.unlink()
+        user_group_ids = sorted(set([group.id for group in user.groups_id]))
+        self.assertEqual(user_group_ids, role1_group_ids)
+        # Remove role1 from the user
```

Comment:
I think it should be `# Remove role1`, not from the user

Reply:
@novawish fixed